pennylane/beta/qnode.py

```diff
@@ -0,0 +1,540 @@
+                f"{list(backprop_devices.keys())} interfaces."
+            )
+
+        raise qml.QuantumFunctionError(
+            f"The {device.short_name} device does not support native computations with "
+            "autodifferentiation frameworks."
+        )
+
+    @staticmethod
+    def _validate_adjoint_method(device):
+        # The conditions below provide a minimal set of requirements that we can likely improve upon in
+        # future, or alternatively summarize within a single device capability. Moreover, we also
+        # need to inspect the circuit measurements to ensure only expectation values are taken. This
+        # cannot be done here since we don't yet know the composition of the circuit.
+
+        supported_device = hasattr(device, "_apply_operation")
+        supported_device = supported_device and hasattr(device, "_apply_unitary")
+        supported_device = supported_device and device.capabilities().get("returns_state")
+        supported_device = supported_device and hasattr(device, "adjoint_jacobian")
+
+        if not supported_device:
+            raise ValueError(
+                f"The {device.short_name} device does not support adjoint differentiation."
+            )
+
+        if device.shots is not None:
+            warnings.warn(
+                "Requested adjoint differentiation to be computed with finite shots."
+                " Adjoint differentiation always calculated exactly.",
+                UserWarning,
+            )
+
+        return "device", {"use_device_state": True, "method": "adjoint_jacobian"}, device
+
+    @staticmethod
+    def _validate_device_method(device):
+        # determine if the device provides its own jacobian method
+        provides_jacobian = device.capabilities().get("provides_jacobian", False)
+
+        if not provides_jacobian:
+            raise qml.QuantumFunctionError(
+                f"The {device.short_name} device does not provide a native "
+                "method for computing the jacobian."
+            )
+
+        return "device", {}, device
+
+    @staticmethod
+    def _validate_parameter_shift(device):
+        model = device.capabilities().get("model", None)
+
+        if model == "qubit":
+            return qml.gradients.param_shift, {}, device
+
+        if model == "cv":
+            return qml.gradients.param_shift_cv, {"dev": device}, device
+
+        raise qml.QuantumFunctionError(
+            f"Device {device.short_name} uses an unknown model ('{model}') "
+            "that does not support the parameter-shift rule."
+        )
+
+    @property
+    def tape(self):
+        """The quantum tape"""
+        return self._tape
+
+    qtape = tape  # for backwards compatibility
+
+    def construct(self, args, kwargs):
+        """Call the quantum function with a tape context, ensuring the operations get queued."""
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat

```

Comment:
Does this need to be updated after that other recent PR where you changed it so that `requires_grad` is returned as `False` when not specified?

Reply:
Yep, this issue still is present in master. Actually, the comment is not very accurate, I should rename it:
```suggestion
            # HOTFIX: to maintain backwards compatibility existing PennyLane code and demos, here we treat
```